tests/test_interactions.py

```diff
@@ -0,0 +1,153 @@
+            if not os.path.exists(directory_path):
+                os.makedirs(directory_path)
+
+            self.driver.save_screenshot(os.path.join(
+                os.path.dirname(__file__),
+                'screenshots',
+                dir_name,
+                name + '.png'
+            ))
+
+        def perform_dragging(x, y, delta_x, delta_y, elem, dir_name='interactions'):
+            """
+            Performs dragging on a node, and return the difference from the start
+            :param x: initial position of the node at the start of the action chain
+            :param y: initial position of the node at the start of the action chain
+            :param delta_x: how much we want to drag the node
+            :param delta_y: how much we want to drag the node
+            :param dir_name: The directory in which we store our screenshots
+            :return: the difference between the position after drag and the starting position
+            """
+            actions.reset_actions()
+            actions.move_to_element_with_offset(
+                self.driver.find_element_by_tag_name('body'), x, y
+            )
+            actions.drag_and_drop_by_offset(source=None, xoffset=delta_x, yoffset=delta_y)
+            actions.click()
+            actions.perform()
+            time.sleep(1)
+
+            elem_json = json.loads(elem.text)
+            new_pos = elem_json.get('renderedPosition')
+            clicked_label = elem_json.get('data', {}).get('label')
+
+            diff_x = round(new_pos['x'] - x)
+            diff_y = round(new_pos['y'] - y)
+
+            save_screenshot(
+                dir_name,
+                'Dragged{}By{}x{}y'.format(clicked_label.replace(' ', ''), diff_x, diff_y)
+            )
+
+            return diff_x, diff_y
+
+        def perform_clicking(x, y, elem, dir_name='interactions'):
+            """
+            :param x: The position on the screen where we want to click
+            :param y: The position on the screen where we want to click
+            :param elem: The element object from where we retrieve the JSON
+            :param dir_name: The directory in which we store our screenshots
+            :return: The label of element most recently clicked, if any
+            """
+            actions.reset_actions()
+            actions.move_to_element_with_offset(
+                self.driver.find_element_by_tag_name('body'), x, y
+            )
+            actions.click()
+            actions.perform()
+
+            time.sleep(1)
+            clicked_label = json.loads(elem.text).get('data', {}).get('label')
+
+            save_screenshot(dir_name, 'Clicked' + clicked_label.replace(' ', ''))
+
+            return clicked_label
+
+        def perform_mouseover(x, y, elem, dir_name='interactions'):
+            actions.reset_actions()
+            actions.move_to_element_with_offset(
+                self.driver.find_element_by_tag_name('body'), x - 100, y
+            )
```

Comment:
dangggg nice work figuring all this stuff out!

Reply:
Thanks haha!